pkg/models/resources/v1alpha3/secret/secrets.go

```diff
@@ -74,5 +80,53 @@ func (s *secretSearcher) filter(object runtime.Object, filter query.Filter) bool
 		return false
 	}
 
+	if filter.Field == query.ParameterFieldSelector {
+		return contains(secret, filter.Value)
+	}
+
 	return v1alpha3.DefaultObjectMetaFilter(secret.ObjectMeta, filter)
 }
+
+// implement a generic query filter to support multiple field selectors with "jsonpath.JsonPathLookup"
+// https://github.com/oliveagle/jsonpath/blob/master/readme.md
+func contains(secret *v1.Secret, fieldSelector query.Value) bool {
+	selectors := strings.Split(string(fieldSelector), ",")
+	for _, selector := range selectors {
+		var negative bool
+		var key string
+		var value string
+		// supports '=', and '!='.(e.g. ?fieldSelector=key1=value1,key2=value2)
+		if strings.Contains(selector, "!=") {
+			negative = true
+			keyValues := strings.Split(selector, "!=")
+			key = keyValues[0]
+			value = keyValues[1]
+		} else {
+			negative = false
+			keyValues := strings.Split(selector, "=")
+			key = keyValues[0]
+			value = keyValues[1]
+		}
```

Comment:
Out of range exception should be caught.

Reply:
> @nuclearwu Out of range exception should be caught.

@wansir Done, review please.